src/GameplaySystem.cpp

```diff
@@ -401,6 +402,26 @@ void GameplaySystem::handle_object_transform_event(const Event& e) {
             )
         );
     }
+
+    glm::vec3 it_pos = glm::vec3(
+                           object_positions_[current_it_id_].x,
+                           object_positions_[current_it_id_].y,
+                           object_positions_[current_it_id_].z
+                       );
+
+    glm::vec3 vec_to = glm::normalize(glm::vec3(it_pos - glm::vec3(x, y, z)));
+    glm::mat4 rot_matrix = glm::toMat4(glm::quat(qw, qx, qy, qz));
```

Comment:
No need to recreate the glm::quat here. You can get it from object_rotations_[object_id].
```
glm::mat4 rot_matrix = glm::toMat4(object_rotations_[object_id]);
```

Reply:
Ah, didn't fix this in the merge conflict. GameplaySystem didn't have a reason to store the rotations before your PR.

Reply:
That's what I thought too. All good :)